grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -807,14 +807,28 @@ class StockMovementApiController {
     }
 
     def requisitionStatusCodes = {
-        def options = RequisitionStatus.listOutboundOptions()?.collect {
-            [
-                    id: it.name(),
-                    value: it.name(),
-                    label: "${g.message(code: 'enum.RequisitionStatus.' + it.name())}",
-                    variant: it.variant.name
-            ]
+        Location currentLocation = Location.get(session.warehouse.id)
+        Boolean isRequestsList = params.getBoolean("isRequestsList")
+
+        List<String> statuses = getRequisitionStatusCodesList(currentLocation.isApprovalRequired(), isRequestsList).collect(mapStatusCodes)
+
+        render([data: statuses] as JSON)
+    }
+
+    // Function for getting appropriate filter options based on current list and supporting requests approval
+    List<RequisitionStatus> getRequisitionStatusCodesList(Boolean hasRequestApprovalActivity, Boolean isRequestsList) {
+        if (isRequestsList) {
+            return hasRequestApprovalActivity ? RequisitionStatus.listRequestOptionsWhenSupportingRequestApproval() : RequisitionStatus.listOutboundOptions()
         }
-        render([data: options] as JSON)
+
+        return hasRequestApprovalActivity ? RequisitionStatus.listOutboundOptionsWhenSupportingRequestApproval() : RequisitionStatus.listOutboundOptions()
```

Comment:
it'd be cleaner if you reverted the statement, so:
```groovy
// If a location doesn't support the request approval, return listOutboundOptions no matter what list is displayed
if (!hasRequestApprovalActivity) {
 return RequisitionStatus.listOutboundOptions()
}
// If supports request approval, check what type of list page it is and return appropriate statuses
if (isRequestList) {
 return RequisitionStatus.listRequestOptionsWhenSupportingRequestApproval()
}
return RequisitionStatus.listOutboundOptionsWhenSupportingRequestApproval()
```

Reply:
I prefer this version + changes in naming that @jmiranda suggested, than moving to the function higher + using if/else. So, I am leaving this part of the code in the version that you suggested, unless there is a strong opinion against it.